### ISSUE TITLE
auras should support hyphenation

### ISSUE THREAD

**Issue by belisarius222** [urbit/arvo#1204]
I've gotten some good mileage out of using incompatible auras in new Ames, like this:

```
+$  bone  @udbone
+$  fragment-num  @udfragmentnum
+$  message-num  @udmessagenum
```

When using long aura names like this, though, they get ugly and hard to read. If I could do kebab-case, it would be more readable:
```
+$  bone  @ud-bone
+$  fragment-num @ud-fragment-num
+$  message-num  @ud-message-num
```

As far as I know, this should be a straightforward change to the aura parser.

**Comment by joshuareagan** [comment]
Should the inclusion of a `hep` change the aura from a non-kebab version? E.g., should `@ud-fragment-num` count as the same aura as `@udfragmentnum`? If not, what should the nesting rule be for kebab auras vs non-kebab auras?

I'd think ignoring the `hep`s as purely cosmetic would be the preferred option (and easier to implement) but maybe there are reasons not to do this.

**Comment by joemfb** [comment]
I agree, ignoring `-` for purposes of aura nesting seems best.

**Comment by philipcmonk** [comment]
I don't know, it seems like if it's there, it's real.  If the heps are purely cosmetic, are `@ud-fragment` and `@udfrag-ment` the same?  Surely not?

**Comment by joemfb** [comment]
They're not the same, but shouldn't they mutually nest? Do you have an alternative idea for a nesting algorithm for hep-allowing auras?

**Comment by Fang-** [comment]
> shouldn't they mutually nest?

I can't think of a reason why they should. Just treating heps as other characters in auras are currently treated (ie "there, therefor real") seems most intuitive. You should be consistent about aura names anyway, there's no reason for `@ud-cool-car-counter` to nest in `@ud-coolcar`.

**Comment by joshuareagan** [comment]
@Fang-, does that mean `@----` counts as an aura?

**Comment by Fang-** [comment]
I imagine we'll want to apply `%term` parsing rules to discourage such insanity.

**Comment by joshuareagan** [comment]
I think term parsing rules are substantially different -- they have numbers at the end, and no capital letters. Auras don't allow numbers, I don't believe, and need capital letters for bitwidth.

**Comment by joshuareagan** [comment]
Perhaps you just meant: (1) no starting an aura with `-`, (2) no ending an aura with `-`, and (3) no more than one consecutive `-`.

**Comment by Fang-** [comment]
Right, I forgot about those differences. Your rules relating to `-` seem fair, except that maybe we want to allow starting with `-`, for named plain `@`? ie `@-private-key`, so that it doesn't overlap with `@p`

**Comment by joemfb** [comment]
@Fang, I think the right thing to do with that example would be `@x-private-key` or something. I don't like the idea of "the empty aura, but with commentary".

**Comment by Fang-** [comment]
What don't you like about it? I'm not sure how it's different from "the p aura, but with commentary". Adding characters that are neither real aura nor commentary seems Not Correct.

But I don't feel super strongly.

**Comment by joemfb** [comment]
From what I understood, the consensus here was that there should be no distinction between "real aura" and "commentary" -- aura nesting should follow a simple prefix match as it currently does, and `@poodle` is distinct from `@p-oodle`. So "empty with commentary" would be nonsensical.
